catalog/spatial/stanag4559/spatial-stanag4559-source/src/test/java/org/codice/ddf/spatial/ogc/stanag4559/catalog/source/data/DAGGenerator.java

```diff
@@ -0,0 +1,440 @@
+            graph.addVertex(attribute);
+            graph.addEdge(node, attribute);
+            attribute = constructAttributeNode("numberOfBands", 1, orb);
+            graph.addVertex(attribute);
+            graph.addEdge(node, attribute);
+            break;
+
+        case "NSIL_MESSAGE":
+            attribute = constructAttributeNode("recipient", "DDF123", orb);
+            graph.addVertex(attribute);
+            graph.addEdge(node, attribute);
+            attribute = constructAttributeNode("messageBody", "This is a message", orb);
+            graph.addVertex(attribute);
+            graph.addEdge(node, attribute);
+            attribute = constructAttributeNode("messageType", "XMPP", orb);
+            graph.addVertex(attribute);
+            graph.addEdge(node, attribute);
+            break;
+
+        case "NSIL_VIDEO":
+            attribute = constructAttributeNode("Category", "IR", orb);
+            graph.addVertex(attribute);
+            graph.addEdge(node, attribute);
+            attribute = constructAttributeNode("encodingScheme", "264ON2", orb);
+            graph.addVertex(attribute);
+            graph.addEdge(node, attribute);
+            break;
+        }
+    }
+
+    /**
+     * Constructs a NSIL_ASSOCIATION subgraph with all optional nodes, as well as all mandatory attributes
+     * for these nodes.  A NSIL_PRODUCT can contain 0...n NSIL_ASSOCIATIONS.  All NSIL_ASSOCIATIONS contain
+     * an edge to the NSIL_CARD node.  A NSIL_ASSOCIATION can contain 1...n NSIL_DESTINATIONS.
+     *
+     * @param NSIL_PRODUCT    - a reference to the root node to link to the DAG graph
+     * @param NSIL_CARD       - a reference to the NSIL_CARD to link to the NSIL_ASSOCIATION subgraph
+     * @param orb             - a reference to the orb to create UCO objects
+     * @param graph           - the graph representation of the DAG
+     * @param numDestinations - the number of NSIL_DESTINATION nodes to create
+     */
+    private static void constructNSIL_ASSOCIATION(Node NSIL_PRODUCT, Node NSIL_CARD, ORB orb,
+            Graph<Node, Edge> graph, int numDestinations) {
+
+        List<String> association_nodes = Arrays.asList("NSIL_RELATION", "NSIL_SOURCE");
+        List<Node> nodePartNodes = getEntityListFromStringList(association_nodes, orb);
+
+        Node root = constructEntityNode("NSIL_ASSOCIATION", orb);
+        graph.addVertex(root);
+        graph.addEdge(NSIL_PRODUCT, root);
+        graph.addEdge(root, NSIL_CARD);
+
+        for (Node n : nodePartNodes) {
+            graph.addVertex(n);
+        }
+
+        graph.addEdge(root, nodePartNodes.get(1));
+
+        graph.addEdge(nodePartNodes.get(1), NSIL_CARD);
+
+        for (int i = 0; i < numDestinations; i++) {
+            Node NSIL_DESTINATION = constructEntityNode("NSIL_DESTINATION", orb);
+            graph.addVertex(NSIL_DESTINATION);
+            graph.addEdge(root, NSIL_DESTINATION);
+            graph.addEdge(NSIL_DESTINATION, NSIL_CARD);
+        }
```

Comment:
Not sure on the project preference where we want variables to look like the spec or to follow standard java coding standards. Normally CAPS with underscore are reserved for statically defined variables.